core/src/main/java/com/google/cloud/sql/core/SqlAdminApiFetcher.java

```diff
@@ -98,104 +88,12 @@ private String generatePublicKeyCert(KeyPair keyPair) {
-              logger.log(Level.SEVERE, msg, throwable);
-            }
-          }
-        };
-    for (ListenableFuture<?> future : futures) {
-      Futures.addCallback(future, runWhenInputAreComplete, executor);
-    }
-
-    return taskFuture;
-  }
-
-  ListenableFuture<InstanceData> getInstanceData(
-      CloudSqlInstanceName instanceName,
-      OAuth2Credentials credentials,
-      AuthType authType,
-      ListeningScheduledExecutorService executor,
-      ListenableFuture<KeyPair> keyPair) {
-    ListenableFuture<Metadata> metadataFuture =
-        executor.submit(() -> fetchMetadata(instanceName, authType));
-    ListenableFuture<Certificate> ephemeralCertificateFuture =
-        whenAllSucceed(
-            () ->
-                fetchEphemeralCertificate(
-                    Futures.getDone(keyPair), instanceName, credentials, authType),
-            executor,
-            keyPair);
-    // Once the API calls are complete, construct the SSLContext for the sockets
-    ListenableFuture<SslData> sslContextFuture =
-        whenAllSucceed(
-            () ->
-                createSslData(
-                    Futures.getDone(keyPair),
-                    Futures.getDone(metadataFuture),
-                    Futures.getDone(ephemeralCertificateFuture),
-                    instanceName,
-                    authType),
-            executor,
-            keyPair,
-            metadataFuture,
-            ephemeralCertificateFuture);
-    // Once both the SSLContext and Metadata are complete, return the results
-    return whenAllSucceed(
-        () -> {
-
-          // Get expiration value for new cert
-          Certificate ephemeralCertificate = Futures.getDone(ephemeralCertificateFuture);
-          X509Certificate x509Certificate = (X509Certificate) ephemeralCertificate;
-          Date expiration = x509Certificate.getNotAfter();
-
-          if (authType == AuthType.IAM) {
-            expiration =
-                getTokenExpirationTime(credentials)
-                    .filter(tokenExpiration -> x509Certificate.getNotAfter().after(tokenExpiration))
-                    .orElse(x509Certificate.getNotAfter());
-          }
-
-          return new InstanceData(
-              Futures.getDone(metadataFuture), Futures.getDone(sslContextFuture), expiration);
-        },
-        executor,
-        metadataFuture,
-        sslContextFuture);
-  }
-
-  private Optional<Date> getTokenExpirationTime(OAuth2Credentials credentials) {
-    return Optional.ofNullable(credentials.getAccessToken().getExpirationTime());
-  }
-
   public String getApplicationName() {
     return apiClient.getApplicationName();
   }
 
   /** Fetches the latest version of the instance's metadata using the Cloud SQL Admin API. */
-  private Metadata fetchMetadata(CloudSqlInstanceName instanceName, AuthType authType) {
+  Metadata fetchMetadata(CloudSqlInstanceName instanceName, AuthType authType) {
```

Comment:
Moved to package so that they can be called by `CloudSqlInstance.fetchInstanceData()`